src/controllers/auth.controller.js

```diff
@@ -0,0 +1,163 @@
+'use strict';
+
+const { mailService } = require('../services/mail.service.js');
+const { jwtService } = require('../services/jwt.service.js');
+const { userService } = require('../services/user.service.js');
+const { ApiError } = require('../utils/api.error.js');
+const { tokenService } = require('../services/token.service.js');
+const { validators } = require('../utils/validators.js');
+
+const register = async (req, res, next) => {
+  const { name, email, password } = req.body;
+
+  const activationToken = jwtService.generateToken();
+
+  await userService.create({
+    name,
+    email,
+    password,
+    activationToken,
+  });
+
+  await mailService.sendActivationMail(email, activationToken);
+
+  res
+    .status(201)
+    .send({ message: 'User created check email to activate account' });
+};
+
+const sendAuth = async ({ id, email, name }, res) => {
+  const userDataToShow = {
+    id,
+    email,
+    name,
+  };
+  const refreshToken = jwtService.createRefreshToken(userDataToShow);
+  const accessToken = jwtService.createAccessToken(userDataToShow);
+
+  const { refreshToken: createdRefreshToken }
+    = await tokenService.createRefreshToken(id, refreshToken);
+
+  res.cookie('refreshToken', createdRefreshToken, {
+    maxAge: 30 * 24 * 60 * 60 * 1000,
+    httpOnly: true,
+    sameSite: 'none',
+    secure: true,
+  });
+
+  res.send({
+    accessToken,
+    user: userDataToShow,
+  });
+};
+
+const activate = async (req, res, next) => {
+  const { activationToken } = req.params;
+  const user = await userService.findByToken(
+    'activationToken',
+    activationToken
+  );
+
+  if (user === null) {
+    throw ApiError.NotFound();
+  }
+
+  const activatedUser = await userService.consumeToken('activationToken', user);
+
+  sendAuth(activatedUser, res);
+};
+
+const login = async (req, res, next) => {
+  const { email, password } = req.body;
+  const user = await userService.findActiveUser(email);
```

Comment:
Consider responding with an error code if the user is not found

Reply:
I already handle it inside of `findActiveUser` - 

`  if (!user) {
    throw ApiError.NotFound();
  }

  if (user.activationToken !== null) {
    throw ApiError.Unauthorized('Not active. Please check your email!');
  }`